### PR TITLE
Add new-line character at the end of JSON

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

### What does this PR do?
After building your che-theia, you may notify that `generator/tests/init-sources/assembly-example/configs/root-compilation.tsconfig.json` file does not have new line character at the end.

Screenshot from 2020-11-05 12-43-42

It's because the file is being overwritten by https://github.com/eclipse/che-theia/blob/master/generator/src/init-sources.ts#L120, without adding new-line character at the end.

The solution is to add `'\n'` at the end of the file content.

### How to test this PR?

- clone che-theia
- run `yarn` to build
- run `git status`

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: generator/src/init-sources.ts [hunk 1]
```diff
@@ -117,8 +116,7 @@ export class InitSources {
 
         // write it back
         const json = JSON.stringify(parsed, undefined, 2);
-        await fs.writeFile(rootCompilationUnitPath, json);
-
+        await fs.writeFile(rootCompilationUnitPath, json + '\n');
```
Comment (benoitf) [inline generator/src/init-sources.ts:119]:
```suggestion
        await fs.writeFile(rootCompilationUnitPath, `${json}\n`);
```

**Review by benoitf — APPROVED** [review]

**Review by vzhukovs — APPROVED** [review]
👍

**Review by RomanNikitenko — APPROVED** [review]

**Review by azatsarynnyy — APPROVED** [review]
Windows requires a different newline escape sequence - `\r\n`.
But I believe everyone is using it in a container.
Thanks!

**Comment by vitaliy-guliy** [comment]
> Windows requires a different newline escape sequence - `\r\n`.
> But I believe everyone is using it in a container.
> Thanks!

Yep, but to leave the file unchanged we need to have `\r` at the end.
BTW, afaik @tsmaeder is using Windows. I will ask him to check the changes.

**Pull request merged**